### PR TITLE
Adding anaconda dev container

### LINKED ISSUE #31: Make awake_sleep_seconds optional and support None
I have some days in my Garmin Sleep data where I have no restless moments sleeping this results in an error:

daily_sleep_dto.awake_sleep_seconds
Input should be a valid integer [type=int_type, input_value=None, input_type=NoneType]

### PR DESCRIPTION
**Pull request by nzigel**
1. Adding dev container to VSCode to enable local execution.
2. Make `awake_sleep_seconds` optional. Closes #31

### CODE REVIEW

**Comment by matin** [comment]
Thanks. I'll take a look

**Comment by nzigel** [comment]
I have made a second commit to make awake_sleep_seconds optional and support None - I created a new [issue #31](https://github.com/matin/garth/issues/31) to link to this.

File: garth/data/sleep.py [hunk 1]
```diff
@@ -47,9 +47,9 @@ class DailySleepDTO:
     deep_sleep_seconds: int
     light_sleep_seconds: int
     rem_sleep_seconds: int
-    awake_sleep_seconds: int
     device_rem_capable: bool
     retro: bool
+    awake_sleep_seconds: Optional[int] = None
```
Comment (matin) [inline garth/data/sleep.py:52]: makes sense

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+# Copy environment.yml (if found) to a temp location so we update the environment. Also
+# copy "noop.txt" so the COPY instruction does not fail if no environment.yml exists.
+COPY environment.yml* .devcontainer/noop.txt /tmp/conda-tmp/
+RUN if [ -f "/tmp/conda-tmp/environment.yml" ]; then umask 0002 && /opt/conda/bin/conda env update -n base -f /tmp/conda-tmp/environment.yml; fi \
+    && rm -rf /tmp/conda-tmp
+
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
```
Comment (matin) [inline .devcontainer/Dockerfile:11]: Will there be a need for additional OS packages? If not, let's remove this

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/anaconda
```
Comment (matin) [inline .devcontainer/devcontainer.json:2]: No need to include this

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "python --version",
+
+	// Configure tool-specific properties.
+	// "customizations": {},
+
+	// Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
+	// "remoteUser": "root"
+}
```
Comment (matin) [inline .devcontainer/devcontainer.json:27]: Any reason to keep this here?

File: garth/data/sleep.py (hunk 1 above)
Comment (matin) [reply garth/data/sleep.py:52]: since this'll require a release to PyPi, please bump the version to `0.4.40`

**Comment by matin** [comment]
@nzigel are you running the pre-commit hooks? Lint is failing

**Review by matin — APPROVED** [review]
looks good!

**Comment by matin** [comment]
Just rebasing and running checks again. When they pass, I'll merge and release

**Pull request merged**